.github/workflows/r_nightly.yml

```diff
@@ -76,43 +88,42 @@ jobs:
 
-          pkgs <- list.files(art_path, pattern = "r-pkg_*")
-          src_i <- grep("r-pkg_src", pkgs)
+          pkgs <- list.files(art_path, pattern = "r-pkg-*")
+          src_i <- grep("r-pkg-src", pkgs)
           src_pkg <- pkgs[src_i]
           pkgs <- pkgs[-src_i]
           libs <- list.files(art_path, pattern = "r-libarrow*")
 
-          new_names <- sub("r-pkg_", "", pkgs, fixed = T)
-          matches <- regmatches(new_names, regexec("(([a-z]+)-[\\.a-zA-Z0-9]+)_(\\d\\.\\d)-(arrow.+)$", new_names))
+          matches <- regmatches(pkgs, regexec("r-pkg-(([a-z]+)-[\\.a-zA-Z0-9]+)-(\\d\\.\\d)-(arrow.+)$", pkgs))
+
 
           dir.create("repo/src/contrib", recursive = TRUE)
-          file.copy(paste0(art_path, "/", src_pkg), paste0("repo/src/contrib/", sub("r-pkg_src-", "", src_pkg)))
+          file.copy(file.path(art_path, src_pkg), file.path("repo/src/contrib", sub("r-pkg-src-", "", src_pkg)))
           tools::write_PACKAGES("repo/src/contrib", type = "source", verbose = TRUE)
 
           for (match in matches) {
-              path <- paste0("repo/bin/", match[[3]], "/contrib/", match[[4]])
+              path <- file.path("repo/bin", match[[3]], "contrib", match[[4]])
               path <- sub("macos", "macosx", path)
               dir.create(path, recursive = TRUE)
-              file <- list.files(art_path, pattern = paste0(match[[1]], "*"), full.names = TRUE)
+              file <- file.path(art_path, match[[1]])
 
-              file.copy(file, paste0(path, "/", match[[5]]))
+              file.copy(file, file.path(path, match[[5]]))
               tools::write_PACKAGES(path, type = paste0(substring(match[[3]], 1, 3), ".binary"), verbose = TRUE)
           }
 
-          lib_names <- sub("r-libarrow-", "", libs)
-          lib_match <- regmatches(lib_names, regexec("([[:alpha:]]+)-(.+)", lib_names))
-
-          dir.create("repo/libarrow/bin/windows", recursive = TRUE)
-          dir.create("repo/libarrow/bin/centos-7", recursive = TRUE)
-          dir.create("repo/libarrow/bin/ubuntu-18.04", recursive = TRUE)
+          lib_match <- regmatches(libs, regexec("r-libarrow-(windows|[[:alpha:]]+-\\d.*)-(arrow-.+)", libs))
 
           for (match in lib_match) {
-              path <- list.files("repo//libarrow//bin", pattern = match[[2]], full.names = TRUE)
-              file <- paste0(art_path, "/r-libarrow-", match[[1]])
-              file.copy(file, paste0(path, "//", match[[3]]))
+              path <- file.path("repo/libarrow/bin", match[[2]])
+              file <- file.path(art_path, match[[1]])
+              dir.create(path, recursive = TRUE, showWarnings = FALSE)
+              file.copy(file, file.path(path, match[[3]]))
           }
-      - run: ls -R repo
+      - name: Show repo contents
+        if: env.GOT_FILES == 'true'
+        run: ls -R repo
       - name: Upload Files
+        if: env.GOT_FILES == 'true'
         uses: burnett01/rsync-deployments@5.2
         with:
           switches: -avzr

```

Comment:
The workflow is erroring: https://github.com/apache/arrow/runs/6690966333?check_suite_focus=true#step:11:19

You may need more switches to make it create the intermediate subdirectories?

Reply:
I didn't want to work on the root directly, that's why I set the path to `arrow/r`. That does not exist yet, someone with committer access should be able to create it via curl:
```
curl -u yourasfid -X MKCOL 'https://nightlies.apache.org/arrow'
curl -u yourasfid -X MKCOL 'https://nightlies.apache.org/arrow/r'
# and while we are at it we could also prepare the java folder
curl -u yourasfid -X MKCOL 'https://nightlies.apache.org/arrow/java'
```